Arm/Arm.cs

```diff
@@ -0,0 +1,290 @@
+         */
+
+        public float MinRoll;
+        public float MaxRoll;
+        public float MinPitch;
+        public float MaxPitch;
+        public float MinYaw;
+        public float MaxYaw;
+        public float Length;
+
+        public bool Fixed;
+        public float FixedAngle;
+
+        public ArmPart(float MinRoll, float MaxRoll,
+                       float MinPitch, float MaxPitch,
+                       float MinYaw, float MaxYaw,
+                       float L)
+        {
+            this.MinRoll = MinRoll;
+            this.MaxRoll = MaxRoll;
+            this.MinPitch = MinPitch;
+            this.MaxPitch = MaxPitch;
+            this.MinYaw = MinYaw;
+            this.MaxYaw = MaxYaw;
+            Length = L;
+            Fixed = false;
+            FixedAngle = 0.0f;
+        }
+
+        public ArmPart(float FixedAngle, float Length)
+        {
+            this.MinRoll = 0.0f;
+            this.MaxRoll = 0.0f;
+            this.MinPitch = 0.0f;
+            this.MaxPitch = (float)Math.PI * 2.0f;
+            this.MinYaw = 0.0f;
+            this.MaxYaw = 0.0f;
+            this.FixedAngle = FixedAngle;
+            this.Fixed = true;
+            this.Length = Length;
+        }
+
+        public static implicit operator ArmPart((int, int) t)
+        {
+            return new ArmPart((float)t.Item1, (float)t.Item2);
+        }
+
+        public static implicit operator ArmPart((float, float) t)
+        {
+            return new ArmPart((float)t.Item1, (float)t.Item2);
+        }
+
+        public static implicit operator ArmPart((double, double) t)
+        {
+            return new ArmPart((float)t.Item1, (float)t.Item2);
+        }
+
+        public static implicit operator ArmPart((int, int, int, int, int, int, int) t)
+        {
+            return new ArmPart((float)t.Item1, (float)t.Item2, (float)t.Item3, (float)t.Item4, (float)t.Item5,
+                             (float)t.Item6, (float)t.Item7);
+        }
+
+        public static implicit operator ArmPart((double, double, double, double, double, double, double) t)
+        {
+            return new ArmPart((float)t.Item1, (float)t.Item2, (float)t.Item3, (float)t.Item4, (float)t.Item5,
+                  (float)t.Item6, (float)t.Item7);
+        }
+
+        public static implicit operator ArmPart((float, float, float, float, float, float, float) t)
+        {
+            return new ArmPart((float)t.Item1, (float)t.Item2, (float)t.Item3, (float)t.Item4, (float)t.Item5,
+                  (float)t.Item6, (float)t.Item7);
+        }
+    }
+
+    public class Arm
+    {
+        ArmPart[] Params;
+        public float[] CurrentPitches { get; private set; }
+        public float[] CurrentRolls { get; private set; }
+        public float[] CurrentYaws { get; private set; }
+
+        private float[] TempPitches;
+        private float[] TempRolls;
+        private float[] TempYaws;
+
+        private float[] PitchSums;
```

Comment:
c# has a convention where private fields are lowercase and begin with an underscore to reduce confusion with property names.

Reply:
According to [Capitalization Conventions](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions):
> Do not use underscores to differentiate words, or for that matter, anywhere in identifiers.

And from [Names of Type Members](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members):
> DO NOT use a prefix for field names. 

As far as I understand, everything should be PascalCase except for parameter names in methods, which should be camelCase.